controllers/llamastackdistribution_controller.go

```diff
@@ -198,12 +214,109 @@ func (r *LlamaStackDistributionReconciler) reconcileDeployment(ctx context.Conte
+func (r *LlamaStackDistributionReconciler) resolveImage(instance *llamav1alpha1.LlamaStackDistribution) (string, error) {
+	if instance.Spec.Server.Distribution.Name != "" {
+		resolvedImage := imageMap[instance.Spec.Server.Distribution.Name]
+		if resolvedImage == "" {
+			return "", fmt.Errorf("failed to validate distribution name: %s", instance.Spec.Server.Distribution.Name)
+		}
+		return resolvedImage, nil
+	}
+
+	return instance.Spec.Server.Distribution.Image, nil
+}
+
+// buildContainerSpec creates the container specification.
+func (r *LlamaStackDistributionReconciler) buildContainerSpec(instance *llamav1alpha1.LlamaStackDistribution, image string) corev1.Container {
+	container := corev1.Container{
+		Name:      llamav1alpha1.DefaultContainerName,
+		Image:     image,
+		Resources: instance.Spec.Server.ContainerSpec.Resources,
+		Env:       instance.Spec.Server.ContainerSpec.Env,
+	}
+
+	if instance.Spec.Server.ContainerSpec.Name != "" {
+		container.Name = instance.Spec.Server.ContainerSpec.Name
+	}
+
+	port := llamav1alpha1.DefaultServerPort
+	if instance.Spec.Server.ContainerSpec.Port != 0 {
+		port = instance.Spec.Server.ContainerSpec.Port
+	}
+	container.Ports = []corev1.ContainerPort{{ContainerPort: port}}
+
+	// Determine mount path
+	mountPath := llamav1alpha1.DefaultMountPath
+	if instance.Spec.Server.Storage != nil && instance.Spec.Server.Storage.MountPath != "" {
+		mountPath = instance.Spec.Server.Storage.MountPath
+	}
+
+	// Add volume mount for storage
+	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+		Name:      "lls-storage",
+		MountPath: mountPath,
+	})
+
+	return container
+}
+
+// configurePodStorage configures the pod storage and returns the complete pod spec.
+func (r *LlamaStackDistributionReconciler) configurePodStorage(instance *llamav1alpha1.LlamaStackDistribution, container corev1.Container) corev1.PodSpec {
+	podSpec := corev1.PodSpec{
+		Containers: []corev1.Container{container},
+	}
+
+	// Add storage volume
+	if instance.Spec.Server.Storage != nil {
+		// Use PVC for persistent storage
+		podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{
+			Name: "lls-storage",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: instance.Name + "-pvc",
+				},
+			},
+		})
+	} else {
+		// Use emptyDir for non-persistent storage
+		podSpec.Volumes = append(podSpec.Volumes, corev1.Volume{
+			Name: "lls-storage",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	}
+
+	// Add any pod overrides
+	if instance.Spec.Server.PodOverrides != nil {
+		podSpec.Volumes = append(podSpec.Volumes, instance.Spec.Server.PodOverrides.Volumes...)
+		container.VolumeMounts = append(container.VolumeMounts, instance.Spec.Server.PodOverrides.VolumeMounts...)
+		podSpec.Containers[0] = container // Update with volume mounts
+	}
+
+	return podSpec
+}
+
```

Comment:
I think these should be extracted to another file to avoid cluttering `controllers/llamastackdistribution_controller.go`. Also, these don't need to be methods, simple functions should do since `LlamaStackDistributionReconciler` is never accessed.

We should keep `controllers/llamastackdistribution_controller.go` 's content focused on the global core logic and reconcile methods. IMO.